yt/config.py

```diff
@@ -66,74 +76,132 @@
-        cp.read(_OLD_CONFIG_FILE)
-        # NOTE: To avoid having the 'DEFAULT' section here,
-        # we are not passing in ytcfg_defaults to the constructor.
-        new_cp = configparser.ConfigParser()
-        new_cp.add_section("yt")
-        for section in cp.sections():
-            for option in cp.options(section):
-                # We changed them all to lowercase
-                if option.lower() in ytcfg_defaults:
-                    new_cp.set("yt", option, cp.get(section, option))
-                    print(f"Setting {option} to {cp.get(section, option)}")
-        open(_OLD_CONFIG_FILE + ".old", "w").write(f)
-        new_cp.write(open(_OLD_CONFIG_FILE, "w"))
-
-    msg = (
-        "The configuration file {} is deprecated. "
-        "Please migrate your config to {} by running: "
-        "'yt config migrate'"
-    )
-    warnings.warn(msg.format(_OLD_CONFIG_FILE, CURRENT_CONFIG_FILE))
-
-if not os.path.exists(CURRENT_CONFIG_FILE):
-    cp = configparser.ConfigParser()
-    cp.add_section("yt")
+OLD_CONFIG_FILE = os.path.join(CONFIG_DIR, "ytrc")
+GLOBAL_CONFIG_FILE = os.path.join(CONFIG_DIR, "yt.toml")
+LOCAL_CONFIG_FILE = None
+
+if os.path.exists(OLD_CONFIG_FILE):
+    if os.path.exists(GLOBAL_CONFIG_FILE):
+        msg = (
+            "The configuration file {} is deprecated. "
+            "Please manually remove it to suppress this warning."
+        )
+        warnings.warn(msg.format(OLD_CONFIG_FILE, GLOBAL_CONFIG_FILE))
+    else:
+        msg = (
+            "The configuration file {} is deprecated. "
+            "Please migrate your config to {} by running: "
+            "'yt config migrate'"
+        )
+        warnings.warn(msg.format(OLD_CONFIG_FILE, GLOBAL_CONFIG_FILE))
+        sys.exit(1)
+
+
+if not os.path.exists(GLOBAL_CONFIG_FILE):
+    cfg = {"yt": {}}
     try:
-        with open(CURRENT_CONFIG_FILE, "w") as new_cfg:
-            cp.write(new_cfg)
+        with open(GLOBAL_CONFIG_FILE, mode="w") as fd:
+            toml.dump(cfg, fd)
     except IOError:
         warnings.warn("unable to write new config file")
 
 
-class YTConfigParser(configparser.ConfigParser):
-    def __setitem__(self, key, val):
-        self.set(key[0], key[1], val)
+class YTConfig:
+    def __init__(self, defaults=None):
+        if defaults is None:
+            defaults = {}
+        self.config_root = ConfigNode(None)
+
+    def get(self, section, *keys, callback=None, **kwargs):
+        if callback is None:
+            callback = lambda leaf: leaf.value  # noqa: E731
+
+        return self.config_root.get_leaf(section, *keys, callback=callback)
+
+    def get_most_specific(self, section, *keys, **kwargs):
+        use_fallback = "fallback" in kwargs
+        fallback = kwargs.pop("fallback", None)
+        try:
+            return self.config_root.get_deepest_leaf(section, *keys)
+        except KeyError as e:
+            if use_fallback:
+                return fallback
+            else:
+                raise e
```

Comment:
I changed the `e` into `err` but not the `use_fallback`. Otherwise, the function will fail for missing keys with `fallback=None` but not with `fallback="something"`. This behavior is consistent with the behavior of `configparser.ConfigParser.get`.

Reply:
fair enough !